project/frontend/src/components/ResultsMap.js

```diff
@@ -59,23 +68,37 @@ function MapContainer(props) {
     );
   };
 
+  // TODO: add a spec here to explain what's happening.
   const createMarkers = () => {
-    if (!restaurants) {
+    if (restaurants.length === 0) {
       return;
     }
-    let markers = [];
-    const numOfMarkers = Math.min(restaurants.length, 4);
-    for (let i = 0; i < numOfMarkers; i++) {
-      const coords = restaurants[i].key.latLngCoords;
-      const numInList = (i + 1).toString();
-      markers.push(
-        <MarkerIcon
-          lat={coords.lat}
-          lng={coords.lng}
-          id={i}
-          aria-label={'Your #' + numInList + ' Match'}
-        />
-      );
+    const markers = [];
+    let coords = restaurants[currentCardIndex].key.latLngCoords;
+    markers.push(
+      <MarkerIcon
+        lat={coords.lat}
+        lng={coords.lng}
+        id={currentCardIndex}
+        aria-label={'Your #' + currentCardIndex + ' Match'}
+      />
+    );
+    const numOfMarkers = Math.min(restaurants.length, 3);
+    for (let i = 1; i <= numOfMarkers; i++) {
```

Comment:
hmm I don't quite follow why this loop is [1, numMarkers] instead of [0, numMarkers) -- maybe add an explanation if it has to be the former?

Reply:
Right before the for loop I added the current marker (which is the marker corresponding to the restaurant card that the user is on). In the for loop I want to add the 3 markers before the current marker and the 3 markers after so to do this I subtract and add 1, 2, and 3 from the current index and add those restaurant markers which is why I exclude 0. I can put this into a comment now if you think that this logic makes sense!

Reply:
sounds good to me! Is there a reason that it's delta instead of i? just curious

Reply:
oh just for naming clarity for what it represents (imo 'i' here would not be as descriptive of the fact that its actually the # of steps before or after the current index and not all loop indices have to be i :D ) 
if you want to come up with a better name though, feel free to go for it -- delta is just sort of in my head a way to think about diff from a certain point